### ISSUE TITLE
[BUG] Cannot Login after update of Silverblue Rawhide - Login loops back to login screen, same applies for tty2 login attempts

### ISSUE THREAD

**Issue by TheOneandOnlyJakfrost** [fedora-silverblue/issue-tracker#228]
This issue tracker is intended only for Silverblue specific issues. We would like to ask you to try to reproduce the issue on a relevant Fedora Workstation release. If you will be able to reproduce there, then please report it in Red Hat Bugzilla or in upstream (preferred for GNOME projects) and not in this issue tracker.

**Describe the bug**
After attempting an update of Fedora Silverblue Rawhide on December 29, at 10.52 EST, the update failed due to conflicts with layered packages. I then attempted to remove the layered packages with `rpm-ostree reset`, I had no overrides. The reset appeared to work and I was presented with a new commit to login to. At each login attempt, a text message would flash (error?) and the login screen would be presented again, so a loop. I tried to access the terminal at tty2, and this worked. Attempted to login as my user and the loop persisted, I was presented with a login prompt with no error presented. I then tried to rollback to F35, thinking that it could be likely rawhide related, this seemed to complete correctly, but the login loop persisted with the same exact results.

**To Reproduce**
To reproduce, 

1. update Fedora Silverblue Rawhide
2. reboot
3. attempt login

**Expected behavior**
Login would continue as normal.
**Screenshots**
Not applicable as screen looks normal and impossible to capture message text reliably.

**OS version:**
```
State: idle
BootedDeployment:
● fedora:fedora/rawhide/x86_64/silverblue
                   Version: Rawhide.20211212.n.0 (2021-12-12T06:10:33Z)
                BaseCommit: 0e1eebd4dcf1a69e8ca6ea0e3e35bead4e0f53d52ed72fd9f8392fde12128344
              GPGSignature: Valid signature by 53DED2CB922D8B8D9E63FD18999F7CBF38AB71F4
           LayeredPackages: asusctl chafa dutree exa fedora-messaging fish foomatic fzf google-noto-emoji-fonts gstreamer1-plugin-openh264 guake
                            java-11-openjdk-devel.x86_64 java-11-openjdk-jmods.x86_64 java-11-openjdk-static-libs.x86_64 jq lshw nmstate nodejs-devel.x86_64
                            nodejs-diagnostic-language-server.noarch nodejs-docs.noarch nodejs-libs.x86_64 nodejs.x86_64 origin.x86_64 qemu tldr tokei tuned vim
                            vim-enhanced

```

**Additional context**
[jrnlctlout.txt](https://github.com/fedora-silverblue/issue-tracker/files/7793760/jrnlctlout.txt)

Output of `journalctl -b -4 --priority=3`

**Comment by TheOneandOnlyJakfrost** [comment]
I have recovered from this issue by doing the following

1. rpm-ostree cleanup -p > to clear out pending transactions
2. rpm-ostree cleanup -b 
3. then rpm-ostree cleanup -m
Afterwards, I could update once I uninstalled three layered packages , vim vim-enhanced and chafa. 
Closing this issue as resolved.

**Comment by travier** [comment]
It is likely that you have changed your default shell to fish(?) and when you reset everything, your password entry was thus invalid.

In general I think you should never change the default user shell and use a test+exec with a fallback in `.bashrc` (or similar) instead.

**Comment by TheOneandOnlyJakfrost** [comment]
Thank you for making that connection. The password entry issue came about prior to the attempt at reset, I had first tried to layer a package needed by me, and couldn't due to a dependancy error with Vim, Vim-enhanced, and chafa. I thought it was likely due to a rawhide related fact that not every package is in sync with changes introduced in rawhide. As I use Vim anyway fairly regularly, I thought to rebase to F35 as a stopgap until those packages settled in rawhide. The rebase was successful, but I couldn't log in due to the looping issue I noted. I don't see how it relates to the choice of shell, I have run Bash (the default) ZSH, and Fish shells throughout my use of Fedora Linux and being on Silverblue since F27 doing the same. this problem has not come up till now. Anyway, I lived with booting into the working commit for a bit, but still wanted to update. So I did a cleanup of everything rpm-ostree, the base, metadata and pending and I was able to update, but only after uninstalling vim, vim-enhanced and chafa. I think, for what it is worth at this moment, I had some issue actually in the commit I was able to login to, that had an effect on the resulting new commit. Likely one of the cleanup commands, not -p, was responsible for helping with my recovery. I can still boot into the working commit I had but not the subsequent looping ones, though I think I could recreate from that dubious commit.
